src/main/java/kr/pickple/back/crew/service/CrewMemberService.java

```diff
@@ -112,4 +112,25 @@ private Member findMemberById(final Long memberId) {
         return memberRepository.findById(memberId)
                 .orElseThrow(() -> new MemberException(MEMBER_NOT_FOUND, memberId));
     }
+
+    @Transactional
+    public void deleteMemberShip(final Long crewId, final Long memberId) {
+        final Crew crew = findByExistCrew(crewId);
+        final Member member = findMemberById(memberId);
+        //TODO:추후, 크루장인지 검증 로직 추가(11월 2일, 소재훈)
+
+        final Optional<CrewMember> crewMember = crewMemberRepository.findByMemberAndCrew(member, crew);
+
+        if (!crewMember.isPresent()) {
+            throw new CrewException(CREW_MEMBER_STATUS_NOT_FOUND, member.getId());
+        }
+
+        final CrewMember deleteCrewMember = crewMember.get();
+
+        if (deleteCrewMember.getStatus() != WAITING) {
+            throw new CrewException(CREW_MEMBER_ALREADY_CONFIRMED, member.getId());
+        }
```

Comment:
`p3`

코드가 많아지면 많아질 수록 복잡해지니 이런식으로 도메인에서 예외를 처리할 수 있도록 넘겨주어 서비스 레이어의 복잡성을 줄여주는 것도 좋을 것 같습니다! 이 부분에 대해서는 어떻게 생각하시나요?

```suggestion
       crewMember.validateCrewMemberIsWaiting();
```

Reply:
이 부분도 WAITING, CONFIRMED 상관없이 삭제하는 로직이 된다면, 그냥 예외처리 자체가 없어도 될 것 같네요! 관련해서 저는 위에서 말씀 드렸습니다~

Reply:
네~ 그래서 해당 부분 삭제하여 처리했습니다~